test/integration/TextEditorTest.kt

```diff
@@ -16,52 +16,52 @@
-import com.vaticle.typedb.studio.test.integration.Utils.PLUS_ICON_STRING
-import com.vaticle.typedb.studio.test.integration.Utils.SAMPLE_DATA_PATH
-import com.vaticle.typedb.studio.test.integration.Utils.CHEVRON_UP_ICON_STRING
+import com.vaticle.typedb.studio.state.common.util.Label
+import com.vaticle.typedb.studio.test.integration.common.Delays
+import com.vaticle.typedb.studio.test.integration.common.Paths.DATA_FILE_NAME
+import com.vaticle.typedb.studio.test.integration.common.Paths.QUERY_FILE_NAME
+import com.vaticle.typedb.studio.test.integration.common.Paths.SAMPLE_DATA_PATH
+import com.vaticle.typedb.studio.test.integration.common.Paths.SCHEMA_FILE_NAME
+import com.vaticle.typedb.studio.test.integration.common.Paths.TQL_DATA_PATH
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.clickIcon
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.clickText
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.connectToTypeDB
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.createData
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.createDatabase
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.delayAndRecompose
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.assertNodeNotExistsWithText
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.assertNodeExistsWithText
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.openProject
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.verifyDataWrite
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.writeDataInteractively
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.writeSchemaInteractively
+import com.vaticle.typedb.studio.test.integration.common.TypeDBRunners.withTypeDB
 import java.io.File
 import kotlin.test.assertTrue
+import kotlinx.coroutines.runBlocking
 import org.junit.Test
 
 class TextEditorTest: IntegrationTest() {
 
     @Test
     fun makeAFileAndSaveIt() {
-        studioTest(composeRule) {
-        // We have to open a project to enable the '+' to create a new file.
-            val path = cloneAndOpenProject(composeRule, source = SAMPLE_DATA_PATH, destination = testID)
+        runBlocking {
+            val path = createData(source = SAMPLE_DATA_PATH, destination = testID)
+            openProject(composeRule, projectDirectory = testID)
 
-            composeRule.onNodeWithText(PLUS_ICON_STRING).performClick()
+            clickIcon(composeRule, Icon.Code.PLUS)
             delayAndRecompose(composeRule)
 
             // This sets saveFileDialog.file!! to the current file, so even though we can't see the window it is useful.
-            composeRule.onNodeWithText(SAVE_ICON_STRING).performClick()
+            clickIcon(composeRule, Icon.Code.FLOPPY_DISK)
```

Comment:
I do now see what you mean in the sense that `clickIcon(FLOPPY_DISK)` does not read as nicely as, say, `clickIcon(SAVE)` would.

To me, this boils down to the `Icon.Code` object's missing functionality, of providing a purpose-based accessor for icons - as opposed to just a shape-based one. That is to say, we could reimplement `Icon.Code` such that it exposes:

```kt
val iconCode1 = Icon.Code.SAVE
```
My understanding is that we currently store icons (from Font Awesome) by registering them under their Font Awesome name, in this case `FLOPPY_DISK`. This makes sense, as it is a unique identifier and is very clear about what the icon looks like. But in terms of usability within the application, it would be useful to introduce a bridge layer between shapes and operations: a one-to-many mapping from shapes to operations.

Now, how would we go about this? Simply adding the operation codes into `Icon.Code`? Refactoring `Icon.Code` such that the shapes are private, possibly called something like `Icon.FontAwesome`, and the operations are public, taking over what is currently named `Icon.Code`?

It's not just `FLOPPY_DISK` that isn't a perfect name. For example, `CHEVRONS_DOWN` would be far more elegant exposed as `EXPAND_ALL`.

This seems like a significantly better solution than aliasing them only in tests, as it could improve clarity in the Studio application too. However, it would overinflate the scope of this PR. How about submitting a follow-up PR for it... or just raising an issue?

Reply:
I agree, glad we came back to this. Raised:
 - https://github.com/vaticle/typedb-studio/issues/626